core/network/src/protocol/sync.rs

```diff
@@ -90,16 +91,26 @@ pub trait Context<B: BlockT> {
 }
 
 #[derive(Debug)]
+/// All the data we have about a Peer that we are trying to sync with
 pub(crate) struct PeerSync<B: BlockT> {
+	/// The common number is the block number that is a common point of ancestry for both our chains
+	/// (as far as we know)
 	pub common_number: NumberFor<B>,
+	// FH: Why aren't these two lines replaced with a `pub info: PeerInfo,` ?
```

Comment:
We have the `PeerInfo` data structure to store these two things that we store in several places. Is there any particular reason we don't re-use that? Just too much overhead for too little gain?

Reply:
I think it's ok to have some duplication of information. Sharing them requires making the API more complicated, and dealing with non-obvious questions such as how to inform when we make a change in the struct.

However I don't like the fact that these `PeerInfo` structs are *too public*.